cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -265,10 +265,10 @@ predicate adjacentDefRead(DefOrUse defOrUse1, UseOrPhi use) {
   exists(IRBlock bb1, int i1, SourceVariable v |
     defOrUse1.asDefOrUse().hasIndexInBlock(bb1, i1, v)
   |
-    exists(IRBlock bb2, int i2 |
-      adjacentDefRead(_, pragma[only_bind_into](bb1), pragma[only_bind_into](i1),
-        pragma[only_bind_into](bb2), pragma[only_bind_into](i2))
-    |
+    exists(IRBlock bb2, int i2, Definition def |
+      adjacentDefRead(pragma[only_bind_into](def), pragma[only_bind_into](bb1),
+        pragma[only_bind_into](i1), pragma[only_bind_into](bb2), pragma[only_bind_into](i2)) and
+      def.getSourceVariable() = v and
```

Comment:
Why wasn't this here before?

Reply:
I thought it wasn't needed because of the conditions on the `def/use` and the `use` (i.e., they are both attached to the same variable). This was actually pointed out by Tom in the initial PR [here](https://github.com/github/codeql/pull/10190#discussion_r960524653), but I managed to convince him that this was actually okay.